api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationCustom.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.api.kafka.model.CertSecretSource;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.GenericSecretSource;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configures a listener to use mutual TLS authentication.
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaListenerAuthenticationCustom extends KafkaListenerAuthentication {
+
+    public static final String FORBIDDEN_PREFIXES = "ssl.keystore.location, ssl.keystore.password, ssl.keystore.type";
```

Comment:
Should we ban these prefixes, as they're generated by the parent Listener class?  This currently doesn't do anything, but wanted to add for discussion. 

Ultimately I _think_ we do want to do this, as we want to control the public/private key the same way as the other listeners. Although, this may be cumbersome for some users, as they may want full control over these configuration values.

Reply:
Yup, my only hesitation here is that then we should add the `tlsTrustedCertificates` here, as otherwise users cannot configure their truststore at all for their listener.

Reply:
My ignorance surrounding `But the broker truststore should be IMHO configured through the standard Strimzi mechanism` is probably what's leading to my confusion. Currently I only see `brokerCertChainAndKey` which helps a user get their key/cert into the keystore, but I do not see anything for a truststore.

Reply:
Well, not directly. But it is controlled using the Clients CA which influences what is in the truststore.

Reply:
Gotcha, and that means we need to add the following to config for all custom listener types:
```
            writer.println(String.format("listener.name.%s.ssl.truststore.location=/tmp/kafka/clients.truststore.p12", listenerNameInProperty));
            writer.println(String.format("listener.name.%s.ssl.truststore.password=${CERTS_STORE_PASSWORD}", listenerNameInProperty));
            writer.println(String.format("listener.name.%s.ssl.truststore.type=PKCS12", listenerNameInProperty));
```

Reply:
I don't know. That would mean it is using TLS client authentication. That might not be the case - at least not for everyone. I would say that if you want TLS client authentication, you should use `type: tls` and not `type: custom`.

Reply:
Ok, let's leave this out for now then. If this is needed down the road for some other use-case, this is a trivial PR to fix.

Reply:
Agree. We can always add that later if we want. So, I guess that means this should then change to `ssl.`, or?